rosenthal/packages/freedesktop.scm

```diff
@@ -46,3 +46,46 @@
                 (sha256
                  (base32
                   "0f72359fzvh6jzri4fd79m34rwm2r55p2ryq4306wrw7xliafzx0")))))))
+
+(define-public xdg-desktop-portal-hyprland
+  (package
+    (name "xdg-desktop-portal-hyprland")
+    (version "0.4.0")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                     (url "https://github.com/hyprwm/xdg-desktop-portal-hyprland")
+                     (commit (string-append "v" version))))
+              (file-name (git-file-name name version))
+              (sha256
+                (base32
+                  "1dcglmx02j73qbmw9qsacamn8byakyzwknpqjnhsyphixb4crrdg"))))
+    (build-system meson-build-system)
+    (arguments
+      (list
+        #:configure-flags #~(list "-Dsystemd=disabled")))
+    (native-inputs (list cmake pkg-config wayland hyprland-protocols))
+    (inputs (list elogind hyprland-protocols pipewire wayland-protocols wayland libinih `(,util-linux "lib")))
+    (home-page "https://github.com/hyprwm/xdg-desktop-portal-hyprland")
+    (synopsis "@code{xdg-desktop-portal} backend for Hyprland")
+    (description
+      "This package provides @code{xdg-desktop-portal-hyprland}. This project extends @code{xdg-desktop-portal-wlr} for Hyprland, adding support for @code{xdg-desktop-portal} screenshot and casting interfaces while adding a few extra portals specific to Hyprland: mostly for window sharing. Requires @code{hyprland-share-picker}.")
+    (license license:expat)))
+
+(define-public hyprland-share-picker
+  (package
+    (inherit xdg-desktop-portal-hyprland)
+    (name "hyprland-share-picker")
+    (build-system qt-build-system)
+    (inputs (modify-inputs (package-inputs xdg-desktop-portal-hyprland)
+                           (append qtwayland-5 slurp)))
```

Comment:
1. IIUC QT6 is required, so qtwayland-5 could be replaced to qtwayland.  
2. slurp (and grim) are only used at runtime, putting it to inputs won't help.  You can put them to propagated-inputs (not recommended), [wrap the program](<https://guix.gnu.org/en/manual/devel/en/html_node/Build-Utilities.html#Wrappers>) to find them, or simply leave the package as-is.

Reply:
That's what I thought too, however using qtwayland caused QT to miss the wayland backend, forcing the window to render with xcb (xwayland). No idea why, as I though this was QT6 as well, but the QT5 wayland seemed to fix it. Open to any ideas as to why this could be.

Reply:
And for the inputs, I'll look into wrapping the program or leaving it. Any reason using propagated-inputs is not recommended?

Reply:
When building a profile, they'll be pulled into it implicitly.  Collisions would happen if building a profile with two or more variants of the same package.

Reply:
`qt-build-system` wraps the programs with QT5 by default, you may set the `#:qtbase` argument to the `qtbase` package.